src/js/stores/validators/AppFormValidators.js

```diff
@@ -24,6 +24,22 @@ const AppFormValidators = {
 
   appIdNoWhitespaces: (str) => str.match(/ /g) == null,
 
+  appIdValidChars: (str) => str.match(/[^a-z0-9\-_\.\/]/g) == null,
+
+  appIdWellFormedPath: (str) => {
+    // This RegExp is taken from the ID field explanation described here:
+    // https://mesosphere.github.io/marathon/docs/rest-api.html#post-v2-apps
+    var idMatchRegExp = "^(([a-z0-9]|[a-z0-9][a-z0-9\-]*" +
+      "[a-z0-9])\.)*([a-z0-9]|[a-z0-9][a-z0-9\-]*[a-z0-9])$";
```

Comment:
This RegExp is kind of crazy, but I see some value in keeping it exactly the same as the docs, so let's leave it as it is.